static/app/components/searchQueryBuilder/hooks/useQueryBuilderState.spec.tsx

```diff
@@ -253,3 +259,191 @@ describe('removeWildcardFromToken', () => {
     expect(result).toBe(expected);
   });
 });
+
+describe('replaceFreeTextTokens', () => {
```

Comment:
In addition to these tests, can you write one that tests the full component? That way you can make sure that focus goes to the correct location and that it all works as expected